test/notifications/notification_server_test.exs

```diff
@@ -244,7 +246,7 @@ defmodule Notifications.NotificationServerTest do
       end
     end
 
-    test "broadcasts and logs new notifications for waivers with recognized reason when no vehicle or ghost is associated" do
+    test "broadcasts and logs new notifications for waivers with recognized reason when no vehicle or ghost is associated on selected routes" do
```

Comment:
A good complement to this "on selected routes" addition would be a test that it doesn't send a notification to a user who's not looking at that route. Otherwise, the old code that sent every notification to everybody would still pass all these tests.